.azure-pipelines/templates/run-tests.yml

```diff
@@ -9,19 +9,23 @@ parameters:
   repo: ''
   coverage: true
   force_base_package: false
+  memray: true
+
+# ${{ replace(replace(parameters.memray, 'true', '--memray'), 'false', '') }}
+# is a hack to act like a ternary operator and avoid duplicate all the steps, with or without the option.
 
 steps:
 - ${{ if and(eq(parameters.test, 'true'), eq(parameters.coverage, 'true'), eq(parameters.force_base_package, 'false')) }}:
-  - script: ddev test --cov --junit ${{ parameters.ddtrace_flag }} ${{ parameters.check }}
+  - script: ddev test --cov --junit ${{ replace(replace(parameters.memray, 'true', '--memray'), 'false', '') }} ${{ parameters.ddtrace_flag }} ${{ parameters.check }}
```

Comment:
I know this is very hacky. At first, I tried something like that:

```yaml 
variables:
  - name: memray_option
    ${{ if eq( parameters['memray'], 'true') }}:
      value: "--memray"
    ${{ else }}:
      value: ""
```

And use `variables.memray_option` but we can not mix parameters, steps and variables in a template. I do not really want to add a bunch of other  `- ${{ if` steps to handle this parameter, which would make the pipeline difficult to read and maintain

Reply:
Would something like setting an environment variable on a previous step (to `--memray`) work here to simplify? Seeing this `replace(replace(...` everywhere _is_ pretty ugly. Oh, the pains of YAML as an ad-hoc programming language... 😞

Reply:
I agree, that's ugly, but still less than copy/pasting all the blocks once more.

Using an env variable could work, that's a cool idea, I need to test that because I'm not familiar enough with azure pipelines to know if it does.

Reply:
I tried something different, let me know what you think :)

Reply:
I had thought of that too, and I agree it's convenient. It makes the name of the parameter a bit misleading but oh, well 🤷 we're doing the same with `ddtrace_flag`. By the way, we may `s/memray_option/memray_flag` for consistency with that as well.

Reply:
Good point, changed it